### PR TITLE
feat: update `PATCH /api/v1/datasets/{dataset_id}/records` endpoint to update vectors

### LINKED ISSUE #4069: [FEATURE] Update `PATCH /api/v1/datasets/{dataset_id}/records` bulk endpoint to include vectors in body

### PR DESCRIPTION
**Pull request by gabrielmbmb**
# Description

This PR adds or updates the following:

- Update `BaseElasticAndOpenSearchEngine.index_records` to also index the vectors of the records.
- Update `PATCH /api/v1/dataset/{dataset_id}/records` endpoint to allow updating the `vectors` of the records.
- Update `PATCH /api/v1/records/{record_id}` endpoint to allow updating the `vectors` of the record.

Closes #4069

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [ ] Test A
- [ ] Test B

**Checklist**

- [ ] I added relevant documentation
- [x] I followed the style guidelines of this project
- [x] I did a self-review of my code
- [x] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [x] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -26,6 +26,9 @@ These are the section headers that we use:
 ### Changed
 
 - Updated `POST /api/v1/datasets/:dataset_id/records` endpoint to allow to create records with `vectors` ([#4022](https://github.com/argilla-io/argilla/pull/4022))
+- Update `PATCH /api/v1/datasets/:dataset_id/records` endpoint to allow to update records with `vectors` ([#4062](https://github.com/argilla-io/argilla/pull/4062))
```
Comment (jfcalvo) [inline CHANGELOG.md]:
```suggestion
- Update `PATCH /api/v1/datasets/:dataset_id/records` endpoint to allow to update records with `vectors`. ([#4062](https://github.com/argilla-io/argilla/pull/4062))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -26,6 +26,9 @@ These are the section headers that we use:
...
+- Update `PATCH /api/v1/records/:record_id` endpoint to allow to update record with `vectors` ([#4062](https://github.com/argilla-io/argilla/pull/4062))
```
Comment (jfcalvo) [inline CHANGELOG.md]:
```suggestion
- Update `PATCH /api/v1/records/:record_id` endpoint to allow to update record with `vectors`. ([#4062](https://github.com/argilla-io/argilla/pull/4062))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -26,6 +26,9 @@ These are the section headers that we use:
...
+- Update `BaseElasticAndOpenSearchEngine.index_records` method to also index record vectors ([#4062](https://github.com/argilla-io/argilla/pull/4062))
```
Comment (jfcalvo) [inline CHANGELOG.md]:
```suggestion
- Update `BaseElasticAndOpenSearchEngine.index_records` method to also index record vectors. ([#4062](https://github.com/argilla-io/argilla/pull/4062))
```

File: src/argilla/server/schemas/v1/datasets.py [hunk 4]
```diff
@@ -423,10 +423,19 @@ class RecordGetterDict(GetterDict):
...
             return default
+
         if key == "suggestions" and not self._obj.is_relationship_loaded("suggestions"):
             return default
+
+        if key == "vectors":
+            if self._obj.is_relationship_loaded("vectors"):
+                return {vector.vector_settings.name: vector.value for vector in self._obj.vectors}
+            else:
+                return default
```
Comment (frascuchon) [inline src/argilla/server/schemas/v1/datasets.py]: in which cases this path is executed?

**Review by frascuchon — APPROVED** [review]

**Review by jfcalvo — APPROVED** [review]

**Pull request merged**